File: app/src/main/java/com/saphamrah/customer/data/shared/PreferencesImpl.java
package com.saphamrah.customer.data.shared;

import com.google.gson.reflect.TypeToken;
import com.saphamrah.customer.base.BasePreferences;
import com.saphamrah.customer.utils.FileManager;

import java.io.File;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.Objects;


public class PreferencesImpl implements BasePreferences {

    private File file;
    private Type mapType;

    private String name;

    public PreferencesImpl(String name) {
        this.name = name;
        try {
            this.file = new File(name).getCanonicalFile();
        } catch (IOException e) {
            e.printStackTrace();
        }
        this.mapType = (new TypeToken() {
        }).getType();
    }


    @Override
    public void putString(String key, String value) {
       /* HashMap<String, String> data = new HashMap<String, String>();
        String readed = FileManager.readFileContent(file);
        if (Objects.equals(readed, "")) {
            data.put(key, value);

        } else {
            data = readed.fromJson(mapType);
            data.get(key) = value;
        }

        FileManager.writeToFile(file, data.toJson());*/
    }

    @Override
    public String getString(String key, String value) {
        return null;
    }

    @Override
    public void clear() {
        FileManager.writeToFile(file, "");

    }
}

File: app/src/main/java/com/saphamrah/customer/utils/PreferencesImpl.kt
package com.saphamrah.customer.utils



import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.saphamrah.customer.base.BasePreferences
import java.io.File
import java.lang.reflect.Type


//class PreferencesImpl(name: String) : BasePreferences {
//
//    private val file = File(name).canonicalFile
//    private val mapType = object : TypeToken<HashMap<String, String>>() {}.type
//
//    override fun putString(key: String, value: String) {
//
//        val data: HashMap<String, String>
//        val readed = FileManager.readFileContent(file)
//        if (readed == "") {
//            data = hashMapOf(
//                Pair(key, value)
//            )
//        } else {
//            data = readed.fromJson(mapType)
//            data[key] = value
//        }
//
//        FileManager.writeToFile(file, data.toJson())
//
//    }
//
//    override fun getString(key: String, default: String): String {
//
//        val readed = FileManager.readFileContent(file)
//        val data: HashMap<String, String>
//        return try {
//            data = readed.fromJson(mapType)
//            data[key] ?: default
//        } catch (ex: Exception) {
//            default
//        }
//
//    }
//
//    override fun putInt(key: String, value: Int) {
//        val data: HashMap<String, Int>
//        val readed = FileManager.readFileContent(file)
//        if (readed == "") {
//            data = hashMapOf(
//                Pair(key, value)
//            )
//        } else {
//            data = readed.fromJson(mapType)
//            data[key] = value
//        }
//
//        FileManager.writeToFile(file, data.toJson())
//    }
//
//    override fun getInt(key: String, value: Int): Int {
//        val readed = FileManager.readFileContent(file)
//        val data: HashMap<String, Int>
//        return try {
//            data = readed.fromJson(mapType)
//            data[key] ?: value
//        } catch (ex: Exception) {
//            value
//        }
//    }
//
//    override fun clear() {
//
//        FileManager.writeToFile(file, "")
//
//    }
//
//     private fun <T> String.fromJson(type: Type): T {
//        return Gson().fromJson(this, type)
//    }
//
//     private fun Any.toJson(): String {
//        return Gson().toJson(this)
//    }
//}
